Scripts/Sources/Automation/Eval.swift

```diff
@@ -0,0 +1,329 @@
+            }
+            try jobsFound.forEach {
+                print("🏃🏻 Running job \($0.key)")
+                try $0.value()
+            }
+        }
+        return !jobsFound.isEmpty
+    }
+
+    static func runCommands(_ title: String, commands: () throws -> Void) {
+        do {
+            if !TravisCI.isRunningLocally() {
+                print("travis_fold:start: \(title)")
+            }
+
+            print("ℹ️ \(title)")
+            try commands()
+
+            if !TravisCI.isRunningLocally() {
+                print("travis_fold:end: \(title)")
+            }
+
+            print("🎉 Finished successfully")
+        } catch let CIError.invalidExitCode(statusCode, errorOutput) {
+            print("😢 Error happened: [InsufficientExitCode] ", errorOutput ?? "unknown error")
+            exit(statusCode)
+        } catch let CIError.logicalError(message) {
+            print("😢 Error happened: [LogicalError] ", message)
+            exit(-1)
+        } catch CIError.timeout {
+            print("🕙 Timeout")
+            exit(-1)
+        } catch {
+            print("😢 Error happened [General]")
+            exit(-1)
+        }
+    }
+
+    // MARK: Tasks
+
+    static let jobs: [String: () throws -> Void] = [
+        "prepareForBuild": prepareForBuild,
+        "prepareExamplesForBuild": prepareExamplesForBuild,
+        "build": build,
+        "buildExamples": buildExamples,
+        "runTests": runTests,
+        "runTestsOnExamples": runTestsOnExamples,
+        "runLinter": runLinter,
+        "generateDocs": generateDocs,
+        "publishDocs": publishDocs,
+        "runCocoaPodsLinter": runCocoaPodsLinter,
+        "testCoverage": testCoverage,
+        "runDanger": runDanger
+    ]
+
+    static func prepareForBuild() throws {
+        if TravisCI.isRunningLocally() {
+            print("🔦 Install dependencies")
+            try Shell.executeAndPrint("rm -f Package.resolved")
+            try Shell.executeAndPrint("rm -rf .build")
+            try Shell.executeAndPrint("rm -rf build")
+            try Shell.executeAndPrint("rm -rf Eval.xcodeproj")
+            try Shell.executeAndPrint("bundle install")
+        }
+
+        print("🤖 Generating project file")
+        try Shell.executeAndPrint("swift package generate-xcodeproj --enable-code-coverage")
+    }
+
+    static func build() throws {
+        print("♻️ Building")
+        try Shell.executeAndPrint("swift build", timeout: 120)
+        try Shell.executeAndPrint("xcodebuild clean build -configuration Release -scheme Eval-Package | bundle exec xcpretty --color", timeout: 120)
+    }
+
+    static func runTests() throws {
+        print("👀 Running automated tests")
+        try Shell.executeAndPrint("swift test", timeout: 120)
+        try Shell.executeAndPrint("xcodebuild test -configuration Release -scheme Eval-Package -enableCodeCoverage YES | bundle exec xcpretty --color", timeout: 120)
```

Comment:
<table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/tevelee/Eval/blob/fc973589729700ac4c55183b359a3f3047c6157f/Scripts/Sources/Automation/Eval.swift#L142"></span>Line should be 120 characters or less: currently 165 characters</td>
    </tr>
  </tbody>
</table>